### ISSUE TITLE
Still supported?

### ISSUE THREAD

**Issue by lindseydiloreto** [carnesmedia/pthumb#1]
Hi, is this plugin still supported? I'm having some trouble getting it to work properly.

**Storage Path**

```
~/Users/lindseydiloreto/Sites/industrialassets/iamachinery/public/assets/brochures
```

**Base URL**

```
/assets/brochures
```

Those paths render this:

```
<img src="/assets/brochures/previews/9143e3d0bbd4eb93b5073acae2fd7bf8.png">
```

... but it doesn't seem to be generating the .png file.

Any thoughts?

**Comment by nathancarnes** [comment]
@lindseydiloreto I'm not actively developing the plugin anymore (since I'm not using Craft as part of my day job anymore), but I'm happy to help out if I can.

A few questions:

1) Is `~/Users/lindseydiloreto/Sites/industrialassets/iamachinery/public/assets/brochures/previews` present and writeable? The plugin should handle creating that `previews` directory automatically, but it'd be worth double-checking.

2) Are there any files generated in that same directory? If so, are they zero bytes or actual thumbnails?

3) Do you have ImageMagick installed and in your `PATH`? PThumb uses `convert` under the hood, which is part of ImageMagick.

**Comment by lindseydiloreto** [comment]
@nathancarnes Thanks for the response!

1) Yes, in fact the plugin created the `previews` directory automatically. Permissions of that folder are `775`.

2) Nothing is being created _within_ `previews`... not even a zero-byte file.

3) I definitely do have ImageMagick installed, though I'm not positive the `PATH` is correct. Where would I look for that information?

On a semi-related note... How would you feel about me spinning off this plugin into something that I would release and support? I can see a few areas which could use some modernizing, and have some feature requests as well. I know you haven't touched this plugin in almost a year, but I'm not sure what your long-term support plans are.

I'm not trying to step on your toes by asking... Clearly you've put a lot of work into this plugin. But I want to make sure my client's project is using something that will be maintained for years to come, and don't want to burden you if it's not a primary focus of yours.

Thanks again for helping out! :)

**Comment by nathancarnes** [comment]
@lindseydiloreto if you get a path to an executable back when you punch `which convert` into your terminal, you should be all set on ImageMagick. 

If you're interested in taking over development on the plugin, that's fantastic! I'm unlikely to do any development on it in the future, and it could certainly use a lot of cleanup and some new features. 

Feel free to fork the project and work on it, or if you'd prefer I'd be happy to add you as an administrator on this repo so that existing links to the plugin will work.

**Comment by nathancarnes** [comment]
As far as more troubleshooting, have you checked your error logs?

**Comment by lindseydiloreto** [comment]
`which convert` returns `/usr/local/bin/convert`, which does exist. All good in theory?

"error logs"... Excellent idea, I should have thought of that! Is there anything in particular that I should look for in there? I glanced through it, and nothing obvious is jumping out at me.

**Comment by lindseydiloreto** [comment]
If it's alright with you, I'll go ahead and fork this project. Here's my plan, let me know if you object to any of this...
- Fork it
- **Make sure you & this repo are properly credited**
- Clean up any modernization issues
- Add a few new features
- And here's the kicker... Eventually **sell** the plugin.

I've got a full suite of Craft plugins, some for free and others for sale. Since I'm supporting a broad collection of plugins, it's much easier for me to deal with bugs and feature requests on an ongoing basis.

Of course, I don't want to do any of this without your blessing. I recognize how much work has gone into getting the plugin to this phase, so I mean you no disrespect with this proposal.

**Comment by nathancarnes** [comment]
I'd probably look through logs for mentions for `convert` or `pthumb`.

As far as forking it off, just let know once that's done and I'll include a link to your repo on this repo so folks can find your more active version. 

The MIT license allows for sale, so that's up to you. My personal preference would be that some version of the plugin remain open source and under active development (with either a paid support or paid upgrade for certain features), but you're certainly under no obligation. I just really like blended open-source/commercial models like that.

**Comment by lindseydiloreto** [comment]
That sounds great! I'll dig through my logs and see if I can find `convert` or `pthumb`.

I'll ping you with the URL once I've got a version polished up. Most likely I'll sell it for fairly cheap (currently thinking the $10-20 range). As I've learned from my other plugins, support can be very time-consuming... just want to make sure that I'm not investing that much time for free.

I love the idea of the "freemium" model, and agree that the community would benefit from having both! Assuming you keep this version alive, I'll be happy to direct folks back here who just need the minimal solution. :)

Thanks again Nathan, feel free to ping me at any point in the future!
